### PR TITLE
Calculate risk scores for questionnaires/surveys

### PR DESCRIPTION
**Pull request by Vicbi**
# *Calculate risk scores for questionnaires/surveys*

## :gear: Release Notes 
Add functionality to calculate the risk score from questionnaires and visualise variation with time.

## :books: Documentation
Update `README.md` file to include questionnaire-related functionalities.

## :white_check_mark: Testing
Add tests to ensure proper functionality.

## :pencil: Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Review by PSchmiedmayer — APPROVED** [review]
Very nice work @Vicbi, only had a few minor comments that I noted here. After these are addressed you can merge the PR 🎉

File: .github/workflows/publish-to-pypi.yml [hunk 1]
```diff
@@ -90,16 +82,31 @@ jobs:
           pip install hatch toml
 
       - name: Update release version
+        env:
+          INPUT_TAG_NAME: ${{ github.event.inputs.tag_name }}
         run: python scripts/update_version.py
 
+      - name: Verify version update
+        run: cat pyproject.toml
```
[the remark this answers is not in the document]
Comment (Vicbi) [reply .github/workflows/publish-to-pypi.yml:90]: I used it only for debugging purposes. I was planning to remove it after confirming everything works as they should. I don't think it's necessary now 👍

File: .github/workflows/publish-to-pypi.yml [hunk 2]
```diff
@@ -90,16 +82,31 @@ jobs:
...
+
       - name: Update authors
         run: python scripts/update_authors.py
 
       - name: Build package
         run: hatch build
 
+      - name: Check if version already exists on Test PyPI
+        run: |
+          VERSION_EXISTS=$(curl -s https://test.pypi.org/pypi/spezi_data_pipeline/json | jq -r ".releases | has(\"${{ needs.determine_environment.outputs.version }}\")")
```
[the remark this answers is not in the document]
Comment (Vicbi) [reply .github/workflows/publish-to-pypi.yml:97]: That's right! I was meaning to adjust this step; thank you for pointing it out!

**Comment by PSchmiedmayer** [comment]
@Vicbi Thank you for continuing to work on the PR, let me know if you need any additional input on this PR 🚀

**Review by PSchmiedmayer — APPROVED** [review]
Thank you @Vicbi for the additional improvements and test cases!

Only had very minor comments; feel free to merge the PR 🚀

File: SpeziDataPipelineTemplateForSurveys.ipynb
[the remark this answers is not in the document]
Comment (Vicbi) [reply SpeziDataPipelineTemplateForSurveys.ipynb:1]: Good point! I had it in mind, but this one slipped. I will add some terminology in the `README.md`.

[the remark this answers is not in the document]
Comment (Vicbi) [reply SpeziDataPipelineTemplateForSurveys.ipynb:1]: Any remaining terms "survey" were replaced by the term "questionnaire".

**Pull request merged**